tests/Microsoft.Azure.CosmosEventSourcingTests/Stores/EventStoreTests.ReadAggregate.cs

```diff
@@ -63,6 +63,32 @@ private TestAggregate()
         }
     }
 
+    private class TestAggregateNoSequencing : AggregateRoot
+    {
+        public int ReplayedEvents { get; private set; }
+
+        protected override void Apply(DomainEvent domainEvent)
+        {
+            switch (domainEvent)
+            {
+                case ReplayableEvent:
+                    ReplayedEvents++;
```

Comment:
Any race condition concerns with multiple threads calling apply at the same time, and reads of the `ReplayedEvents`?

Reply:
Yes there will be, kind of a long story to this change, I plan to update the docs alongside this as well. But in short, the sequence number and the strong consistency provided by the atomic event is great in a single region scenario.

However, when we have been assessing the ability to enable multi region writes where traffic would be load balanced across two regions both writing to there local cosmos db data store, we could not guarantee that this sequence number remains in sync and in some use cases for event sourcing it doesn't really matter. The plan here is to handle simultaneous writes, as long as that event is valid at the point in time we are going to let it go ahead.

This diagram is covering our use case, but I am still fleshing out all the details of whether this is 100% suitable via some POCs.

<img width="1327" alt="Screenshot 2023-11-02 at 18 05 29" src="https://github.com/IEvangelist/azure-cosmos-dotnet-repository/assets/23740684/ce800ee1-b239-4d65-be5e-6fad47d5a5d6">

Reply:
I'm good with it if you are, let's :shipit: